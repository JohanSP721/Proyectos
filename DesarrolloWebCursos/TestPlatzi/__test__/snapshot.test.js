import { getCharacter } from '../snapshot';
import rick from '../rick.json';

describe('Es hora de las instantáneas', () =>
{
	test('Snapshot', () =>
	{
		expect(getCharacter(rick)).toMatchSnapshot();
	});

	test('Siempre fallara la instantánea', () =>
	{
		const user =
		{
			createAt: new Date(),
			id: Math.floor(Math.random() * 100)
		}

		expect(user).toMatchSnapshot(
		{
			createAt: expect.any(Date),
			id: expect.any(Number)
		});
	});

	test('Tenemos excepciones', () =>
	{
		const user =
		{
			id: Math.floor(Math.random() * 100),
			name: "Oscar Barajas"
		}

		expect(user).toMatchSnapshot(
			{
				id: expect.any(Number)
			});
	});
});