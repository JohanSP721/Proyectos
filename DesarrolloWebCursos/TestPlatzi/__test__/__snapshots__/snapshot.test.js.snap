// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instant치neas Siempre fallara la instant치nea 1`] = `
Object {
	"createAt": Any<Date>,
	"id": Any<Number>,
}
`;

exports[`Es hora de las instant치neas Snapshot 1`] = `
Object {
	"gender": "Male",
	"id": 1,
	"name": "Rick Sanchez",
	"species": "Human",
	"status": "Alive",
}
`;

exports[`Es hora de las instant치neas Tenemos excepciones 1`] = `
Object {
	"id": Any<Number>,
	"name": "Oscar Barajas",
}
`;
